dandiapi/api/views/version.py

```diff
@@ -101,6 +101,9 @@ def publish(self, request, **kwargs):
                 status=status.HTTP_405_METHOD_NOT_ALLOWED,
             )
 
+        if not old_version.valid:
+            return Response('Metadata is not valid', status=status.HTTP_400_BAD_REQUEST)
```

Comment:
It might be better to specify that this can happen if the version metadata _or_ any asset metadata is invalid.